### PR TITLE
[Merged by Bors] - chore: Remove (some) algebraic dependencies to `Data.Nat.Interval`

### PR DESCRIPTION
**Pull request by YaelDillies**
and move `Data.Nat.Interval`/`Data.Fin.Interval` to `Order.Interval.Finset.Nat`/`Order.Interval.Finset.Fin`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Interval/Finset/Fin.lean [hunk 1]
```diff
@@ -3,10 +3,9 @@ Copyright (c) 2021 Yaël Dillies. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 -/
-import Mathlib.Data.Nat.Interval
-import Mathlib.Order.Interval.Finset.Basic
+import Mathlib.Order.Interval.Finset.Nat
 
-#align_import data.fin.interval from "leanprover-community/mathlib"@"1d29de43a5ba4662dd33b5cfeecfc2a27a5a8a29"
+#align_import Order.Interval.Finset.Fin from "leanprover-community/mathlib"@"1d29de43a5ba4662dd33b5cfeecfc2a27a5a8a29"
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Interval/Finset/Fin.lean:8]:
```suggestion
#align_import data.fin.interval from "leanprover-community/mathlib"@"1d29de43a5ba4662dd33b5cfeecfc2a27a5a8a29"
```

File: Mathlib/Order/Interval/Finset/Nat.lean [hunk 2]
```diff
@@ -3,11 +3,9 @@ Copyright (c) 2021 Yaël Dillies. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 -/
-import Mathlib.Order.Interval.Finset.Basic
 import Mathlib.Order.Interval.Multiset
-import Mathlib.Algebra.Order.Interval.Finset
 
-#align_import data.nat.interval from "leanprover-community/mathlib"@"1d29de43a5ba4662dd33b5cfeecfc2a27a5a8a29"
+#align_import Order.Interval.Finset.Nat from "leanprover-community/mathlib"@"1d29de43a5ba4662dd33b5cfeecfc2a27a5a8a29"
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Interval/Finset/Nat.lean:8]:
```suggestion
#align_import data.nat.interval from "leanprover-community/mathlib"@"1d29de43a5ba4662dd33b5cfeecfc2a27a5a8a29"
```

File: Mathlib/Order/Interval/Finset/Nat.lean [hunk 3]
```diff
@@ -345,11 +345,15 @@ theorem range_image_pred_top_sub (n : ℕ) :
...
   rw [Finset.range_eq_Ico, map_eq_image]
-  convert (@Ico_union_Ico_eq_Ico _ _ _ _ _ (a + b) a.zero_le le_self_add).symm
-  exact image_add_left_Ico _ _ _
+  convert (Ico_union_Ico_eq_Ico a.zero_le le_self_add).symm
+  ext x
+  simp only [Ico_zero_eq_range, mem_image, mem_range, addLeftEmbedding_apply, mem_Ico]
+  constructor
+  · aesop
+  · rintro h
+    exact ⟨x - a, by omega⟩
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Interval/Finset/Nat.lean:356]: This doesn't look great. What's the problem here?

Comment (YaelDillies) [reply Mathlib/Order/Interval/Finset/Nat.lean:356]: `image_add_left_Ico` is an algebraic order lemma. I'm just reproving it quickly in the special case of `Nat`.

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Pull request closed without merging**